keyboards/irene/keymaps/default/keymap.c

```diff
@@ -50,111 +52,184 @@ const uint16_t PROGMEM fn_actions[] = {
 
 const macro_t *action_get_macro(keyrecord_t *record, uint8_t id, uint8_t opt)
 {
-  // MACRODOWN only works in this function
-      switch(id) {
+    // MACRODOWN only works in this function
+    switch(id) {
         case 0:
-          if (record->event.pressed) {
-            register_code(KC_RSFT);
-          } else {
-            unregister_code(KC_RSFT);
-          }
-        break;
-      }
+            if (record->event.pressed) {
+                register_code(KC_RSFT);
+            } else {
+                unregister_code(KC_RSFT);
+            }
+            break;
+    }
     return MACRO_NONE;
 };
 
 bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    // Exit idle mode and stop timer once it detects key press
+    if ((record->event.pressed) && (idle_mode == 1)) {
+        idle_mode = 0;
+        timer_loop = 0;
+        lcd_clrscr();
+        timer1_stop();
+    }
+
+    /*
+    Keyboard Layer Logic:
+        Layer 0 (default)
+        Layer 1:
+            Triggered by holding MO(_RS)
+        Layer 2:
+            Triggered by TG(_LW) in layer 1
+            Exit upon TO(_QW)
+    */
     switch (keycode) {
         case MO(_RS):
-	case TO(_QW):
-	    if (record->event.pressed) {
-		if (i == 2)
-		    i = 0;
-		else
-		    i++;
-	    }
-	    else {
-		if ((i != 2) && (i != 0))
-		    i--;
-	    }
-	    break;
-	case TG(_LW):
-	    if (record->event.pressed) {
-	    	i++;
-	    }
-	    break;
-	case KC_NO:
-	    if (record->event.pressed) {
-	    	if (lcd_off) {
-	    	    lcd_init(LCD_DISP_ON);
-		    lcd_clrscr();
-		    lcd_puts("LCD ON");
-	            lcd_off = false;
-	    	}
-	    	else {
-		    lcd_clrscr();
-		    lcd_sleep(YES);
-		    lcd_off = true;
-	    	}
-	    }
-	    return true;
-	default:
-	    return true;
+        case TO(_QW):
+            // Increment layer upon MO(_RS) keypress, return to 0 upon TO(_QW)
+            if (record->event.pressed) {
+                if (keyboard_layer == 2) {
+                    keyboard_layer = 0;
+                }
+                else {
```

Comment:
Please be consistent between with where you place your braces. On line 60 you used:
```
} else {
```
And on this line you use:
```
}
else {
```